vignettes/git_usage.Rmd

```diff
@@ -175,4 +175,53 @@ knitr::include_graphics("github_conflicts.png", dpi = 144)
 * [GitHub and RStudio](https://resources.github.com/whitepapers/github-and-rstudio/)
 * [Happy Git and GitHub for the useR](https://happygitwithr.com/)
 
+# Package Release
+
+## Quarterly Release
+
+A package release is done in four parts:
+
+1) Declare a freeze on all Pull Requests going into the `devel` branch via internal channels.
+1) Create a PR from `devel` into the `pre-release` branch.  Verify that all CI/CD checks are passing and bundle up and send off to [CRAN](https://cran.r-project.org/).
+1) Once package is available on CRAN, another Pull Request is created for merging the `pre-release` branch the `main` branch.  This will trigger the action to rebuild the `{admiral}` website with all the updates for this release.
+1) Use the release button on GitHub to "release" the package onto GitHub.  This release onto Github archives the version of code within the `main` branch, attaches the News/Changelog file, bundles the code into a `tar.gz` file and makes a validation report via the GitHub action `validation` from [insightsengineering/validatoR](https://github.com/insightsengineering/thevalidatoR).  Please see past [admiral releases](https://github.com/pharmaverse/admiral/releases) for reference. 
+
+**Quarterly Release:**  `devel >> pre-release >> main` 
+
+## Hot Fix Release
+
+Occasionally we will need to release a hot fix to address a package breaking bug.  A hot fix release is done in 6 parts:
+
+1) Create two branches called `patch` and `pre-release` off the current `main`.
```

Comment:
Should we protect these branches and do not delete them after merging?

Reply:
I was thinking about that...

For the patch branch, if you patch infrequently then the patch branch might get out of synch and sometime spent figuring out what to merge in.  Better to delete and re-branch off the current main for patches.

For pre-release - this might get tricky again.  We might have a pre-release for patch/hotfix and a pre-release for a quarterly release.  But I guess this should just be a mirror of the main anyways.  

So...maybe pre-release is kept and protected while patches are deleted?

Reply:
Perhaps there is a way to always keep the main and pre-release branch in sync?

Reply:
makes sense.  there is a github action that can help automate this so we don't lose track of it.

https://github.com/marketplace/actions/sync-branches @dinakar29.  Is this a good one to keep our main and patch branch in sync?

Reply:
Looks pretty good to me 👍🏽

Reply:
Created an issue #1332

Reply:
Cool, I'll look into this 👆